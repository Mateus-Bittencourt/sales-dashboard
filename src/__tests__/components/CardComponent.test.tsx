import { CardComponent } from '@components'
import { render } from '@testing-library/react'
import type { Theme } from '@types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@resources/themesList'

describe('CardComponent', () => {
  const renderComponent = (theme: Theme, className?: string) =>
    render(
      <ThemeProvider theme={theme}>
        <CardComponent className={className} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`when using ${name}`, () => {
      it('should match snapshot without any class', () => {
        const { asFragment } = renderComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with success class', () => {
        const { asFragment } = renderComponent(theme, 'success')
        expect(asFragment()).toMatchSnapshot()
      })

      it('should match snapshot with warning class', () => {
        const { asFragment } = renderComponent(theme, 'warning')
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
