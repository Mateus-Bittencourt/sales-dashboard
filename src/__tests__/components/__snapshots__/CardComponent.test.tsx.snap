// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent when using Dark Theme should match snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0f1535;
  border: 0.0625rem solid #21497d;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Dark Theme should match snapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0f1535;
  border: 0.0625rem solid #21497d;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Dark Theme should match snapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0f1535;
  border: 0.0625rem solid #21497d;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Dark Theme should match snapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0f1535;
  border: 0.0625rem solid #21497d;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Light Theme should match snapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Light Theme should match snapshot with success class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 success"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Light Theme should match snapshot with warning class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0 warning"
  />
</DocumentFragment>
`;

exports[`CardComponent when using Light Theme should match snapshot without any class 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #fff;
  border: 0.0625rem solid #e0e0e0;
  border-radius: 0.5rem;
  box-sizing: border-box;
  padding: 1.5rem;
  width: 100%;
}

.c0.alert {
  background-color: #e80000;
  border-color: #e80000;
}

.c0.success {
  background-color: #008000;
  border-color: #008000;
}

.c0.warning {
  background-color: #f7a300;
  border-color: #f7a300;
}

.c0 a {
  color: inherit;
  text-decoration: none;
}

<div
    class="c0"
  />
</DocumentFragment>
`;
